checks/check_extra760

```diff
@@ -24,45 +24,70 @@ CHECK_DOC_extra760='https://docs.aws.amazon.com/secretsmanager/latest/userguide/
+    if [[ $(echo "${LIST_OF_FUNCTIONS}" | grep -E 'AccessDenied|UnauthorizedOperation|AuthorizationError') ]]; then
+      textInfo "${regx}: Access Denied trying to list Lambda functions" "${regx}"
       continue
     fi
-    if [[ $LIST_OF_FUNCTIONS ]]; then
-      for lambdafunction in $LIST_OF_FUNCTIONS;do
-        LAMBDA_FUNCTION_FOLDER="$SECRETS_TEMP_FOLDER/extra760-$lambdafunction-$regx"
-        LAMBDA_FUNCTION_FILE="$lambdafunction-code.zip"
-        LAMBDA_CODE_LOCATION=$($AWSCLI lambda get-function $PROFILE_OPT --region $regx --function-name $lambdafunction --query Code.Location --output text 2>&1)
-        if [[ $(echo "$LAMBDA_CODE_LOCATION" | grep AccessDenied) ]]; then
-          textInfo "$regx: Access Denied trying to get Lambda functions" "$regx" "$lambdafunction"
+    if [[ "${LIST_OF_FUNCTIONS}" != "" && "${LIST_OF_FUNCTIONS,,}" != "none" ]]; then
+      for lambdafunction in ${LIST_OF_FUNCTIONS}; do
+        LAMBDA_FUNCTION_FOLDER="${SECRETS_TEMP_FOLDER}/extra760-${lambdafunction}-${regx}"
+        LAMBDA_FUNCTION_FILE="${lambdafunction}-code.zip"
+        LAMBDA_CODE_LOCATION=$("${AWSCLI}" lambda get-function ${PROFILE_OPT} --region "${regx}" --function-name "${lambdafunction}" --query 'Code.Location' --output text 2>&1)
+        if [[ $(echo "${LAMBDA_CODE_LOCATION}" | grep -E 'AccessDenied|UnauthorizedOperation|AuthorizationError') ]]; then
+          textInfo "${regx}: Access Denied trying to get Lambda functions" "${regx}" "${lambdafunction}"
           continue
         fi
-        mkdir $LAMBDA_FUNCTION_FOLDER
+        
+        mkdir "${LAMBDA_FUNCTION_FOLDER}"
+        
         # DOWNLOAD the code in a zip file
-        curl -s $LAMBDA_CODE_LOCATION -o $LAMBDA_FUNCTION_FOLDER/$LAMBDA_FUNCTION_FILE
-        unzip -qq $LAMBDA_FUNCTION_FOLDER/$LAMBDA_FUNCTION_FILE -d $LAMBDA_FUNCTION_FOLDER
-        FINDINGS=$(secretsDetector folder $LAMBDA_FUNCTION_FOLDER)
-          if [[ $FINDINGS -eq 0 ]]; then
-            textPass "$regx: No secrets found in Lambda function $lambdafunction code" "$regx" "$lambdafunction"
-            # delete files if nothing interesting is there
-            rm -fr $LAMBDA_FUNCTION_FOLDER
+        CURL_ERROR=$(curl -s --show-error "${LAMBDA_CODE_LOCATION}" -o "${LAMBDA_FUNCTION_FOLDER}/${LAMBDA_FUNCTION_FILE}" 2>&1)
+        if [[ "${CURL_ERROR}" != "" ]]; then
+          textInfo "${regx}: Error trying to get Lambda function code for ${lambdafunction} - ${CURL_ERROR}" "${regx}" "${lambdafunction}"
+          # delete files to not leave trace, user must look at the function
+          if [[ -d "${LAMBDA_FUNCTION_FOLDER}" ]]; then
+            rm -fr "${LAMBDA_FUNCTION_FOLDER}"
+          fi
+          continue
+        fi
+        IS_ZIP_FILE=$(file "${LAMBDA_FUNCTION_FOLDER}/${LAMBDA_FUNCTION_FILE}" | grep 'Zip archive data')
+        if [[ "${IS_ZIP_FILE}" == "" ]]; then
```

Comment:
```suggestion
        if [[ -n "${IS_ZIP_FILE}"  ]]; then
```

Reply:
I guess here we need to use `-z`, not `-n`, correct?

Reply:
Yes, that's it because we are checking if `IS_ZIP_FILE` is an empty variable.